controller/api/models.py

```diff
@@ -518,7 +541,25 @@ def _default_scale(self, user, release):
 
         self.scale(user, structure)
 
-    def logs(self, log_lines=str(settings.LOG_LINES)):
+    def _get_etcd_client(self):
+        if self._etcd_client is None:
+            try:
+                self._etcd_client = etcd.Client(host=settings.ETCD_HOST,
+                                                port=int(settings.ETCD_PORT))
+            except etcd.EtcdException:
+                logger.log(logging.WARNING,
+                           'Cannot synchronize with etcd cluster')
+        return self._etcd_client
+
+    def _get_logging_mode(self):
+        try:
+            return self._get_etcd_client().get('/deis/logs/handlertype').value
```

Comment:
ideally we should be reaching out to a setting like it's done [here](https://github.com/deis/deis/blob/master/controller/api/models.py#L95) and [here](https://github.com/deis/deis/blob/master/controller/deis/settings.py#L292), which can be templated by confd [here](https://github.com/deis/deis/blob/master/controller/templates/confd_settings.py). That way we can extract this bit away so we're not reliant as much on etcd if we were to migrate to something else.